controller/cmd/proxy-api/main.go

```diff
@@ -18,6 +18,7 @@ func main() {
 	metricsAddr := flag.String("metrics-addr", ":9996", "address to serve scrapable metrics on")
 	kubeConfigPath := flag.String("kubeconfig", "", "path to kube config")
 	k8sDNSZone := flag.String("kubernetes-dns-zone", "", "The DNS suffix for the local Kubernetes zone.")
+	enableH2Upgrade := flag.Bool("enable-h2-upgrade", true, "Enable transparently upgraded HTTP2 connections among pods in the service mesh")
```

Comment:
Related to @siggy's comment on #1925 -- my preference is for boolean flags to default to false, for no other reason than the fact that if you wanted to actually disable protocol upgrade using this flag, you'd have to specify it as `-enable-h2-upgrade=false`. If you inverted it, then you could disable protocol upgrade with `-disable-h2-upgrade` (no argument). And that would be more consistent with the flag that you're adding in #1926.

Reply:
My intent here was that the service exposes several enable- flags, and it would be inconsistent to expose otherwise. What if the default value changes? Changing the flags is an "API breakage" at that point.